### ISSUE TITLE
HTTPS-sink-connector duplicates protocol 

### ISSUE THREAD

**Issue by OlyaVashchenko** [apache/camel-kafka-connector#1077]
Hello. I'm trying to use **camel-https-kafka-connector** to _POST_ data into _SuccessFactors_ (with basic authentication - username and password) and get the following **error in my console**: 
```
Caused by: org.apache.camel.ResolveEndpointFailedException: Failed to resolve endpoint: https://https:apisalesdemo8.successfactors.com/odata/v2/upsert?httpMethod=POST due to: The uri part is not configured correctly. You have duplicated the http(s) protocol.
        at org.apache.camel.component.http.HttpComponent.createEndpoint(HttpComponent.java:298)
        at org.apache.camel.support.DefaultComponent.createEndpoint(DefaultComponent.java:171)
        at org.apache.camel.impl.engine.AbstractCamelContext.doGetEndpoint(AbstractCamelContext.java:920)
```
My **.properties** file:
```
name=CamelHttpsSinkConnector
connector.class=org.apache.camel.kafkaconnector.https.CamelHttpsSinkConnector
tasks.max=1
key.converter=org.apache.kafka.connect.storage.StringConverter
value.converter=org.apache.kafka.connect.storage.StringConverter
topics=camel-test
camel.sink.path.httpUri=apisalesdemo8.successfactors.com/odata/v2/upsert?processInactiveEmployees=true&purgeType=full
camel.sink.endpoint.httpMethod=POST
#camel.sink.endpoint.authMethod=Basic
#camel.sink.endpoint.authUsername=xxx
#camel.sink.endpoint.authPassword=xxx
```
If I uncomment the authentication part, I get an error of the same nature.
Kafka version: 2.13-2.7.0, connector version: 0.8.0

**Comment by OlyaVashchenko** [comment]
Just in case: I downloaded the connector from **https://camel.apache.org/camel-kafka-connector/latest/connectors.html**

**Comment by tadayosi** [comment]
Looks like this is another issue that EndpointUriFactory causes.

**Comment by oscerd** [comment]
@tadayosi since you worked on other feel free to have a look at this one too. I think the scheme is wrong, but the duplication seems to be a different problem.

**Comment by tadayosi** [comment]
As demonstrated in my itests, the issue occurs when `httpUri` includes additional query string such as `?processInactiveEmployees=true&purgeType=full`. I believe it requires camel-core to be patched.

A quick workaround for the time being would be to use `camel.sink.url` to provide the entire URL yourself.

**Comment by OlyaVashchenko** [comment]
Ok, thanks! The workaround is working.

**Comment by OlyaVashchenko** [comment]
Should I close the issue now? :)

**Comment by tadayosi** [comment]
Great it works for you. It's up to you whether to close it or not now. I'm working on the root cause and trying to fix it anyway.

**Comment by OlyaVashchenko** [comment]
I have to ask: can this issue affect authorization? I've only checked the request in Postman and it works fine with the credentials I provide. However the connector (I launch it via the console) throws me another exception ( org.apache.camel.http.common.HttpOperationFailedException) with statusCode 401.

**Comment by OlyaVashchenko** [comment]
I'll try to make the request both using GET method and launching a pure Camel application later today.

**Comment by tadayosi** [comment]
Yes I think so. Every parameter with prefix `camel.sink.endpoint.*` should be explicitly put in the `camel.sink.url` if you go with the workaround, otherwise those parameters would be ignored (I think).

**Comment by OlyaVashchenko** [comment]
Seems like 3 authentication parameters works fine when explicitly put in the ```camel.sink.url```. At least the 401 error is gone :) However I'm not sure where to place my custom headers (like ```Content-type```). I don't think that adding them to the above property works (I get the 502 error caused by the wrong Http method).

**Comment by OlyaVashchenko** [comment]
Please tell if my questions are "off-topic".

**Comment by tadayosi** [comment]
The issue is fixed with latest camel master, so it can be closed once CKC 0.9.0 is released.

**Comment by tadayosi** [comment]
This is the fix: https://github.com/apache/camel/commit/af773d814b17e8533ead23d3989efdff6340fea9

**Comment by OlyaVashchenko** [comment]
Great, thank you!

**Comment by tadayosi** [comment]
Confirmed fixed with 0.9.0.
